functions/Remove-DbaDbData.ps1

```diff
@@ -0,0 +1,166 @@
+        [switch]$EnableException
+    )
+
+    begin {
+        $null = Test-ExportDirectory -Path $Path
+    }
+    process {
+        if (-not $InputObject -and -not $SqlInstance) {
+            Stop-Function -Message "You must pipe in a database or a server, or specify a SqlInstance"
+            return
+        }
+
+        if ($SqlInstance) {
+            $InputObject = $SqlInstance
+        }
+
+        foreach ($input in $InputObject) {
+            $inputType = $input.GetType().FullName
+            switch ($inputType) {
+                'Sqlcollaborative.Dbatools.Parameter.DbaInstanceParameter' {
+                    Write-Message -Level Verbose -Message "Processing DbaInstanceParameter through InputObject"
+                    $dbDatabases = Get-DbaDatabase -SqlInstance $input -SqlCredential $sqlcredential -Database $Database -ExcludeDatabase $ExcludeDatabase -ExcludeSystem
+                }
+                'Microsoft.SqlServer.Management.Smo.Server' {
+                    Write-Message -Level Verbose -Message "Processing Server through InputObject"
+                    $dbDatabases = Get-DbaDatabase -SqlInstance $input -SqlCredential $sqlcredential -Database $Database -ExcludeDatabase $ExcludeDatabase -ExcludeSystem
+                }
+                'Microsoft.SqlServer.Management.Smo.Database' {
+                    Write-Message -Level Verbose -Message "Processing Database through InputObject"
+                    $dbDatabases = $input | Where-Object { -not $_.IsSystemObject }
+                }
+                default {
+                    Stop-Function -Message "InputObject is not a server or database."
+                    return
+                }
+            }
+
+            foreach ($db in $dbDatabases) {
+                if ($Pscmdlet.ShouldProcess($db.Name, "Removing all data on $($db.Parent.Name)")) {
+                    Write-Message -Level Verbose -Message "Truncating tables in $db on instance $instance"
+                    $instance = $db.Parent
+                    try {
+
+                        # Collect up the objects we need to drop and recreate
+                        $objects = @()
+                        $objects += Get-DbaDbForeignKey -SqlInstance $instance -Database $db.Name
+                        $objects += Get-DbaDbView -SqlInstance $instance -Database $db.Name -ExcludeSystemView
+
+                        # Script out the create statements for objects
+                        $createOptions = New-DbaScriptingOption
+                        $createOptions.Permissions = $true
+                        $createOptions.ScriptBatchTerminator = $true
+                        $createOptions.AnsiFile = $true
+                        $null = $objects | Export-DbaScript -FilePath "$Path\$($db.Name)_Create.Sql" -ScriptingOptionsObject $createOptions
+
+                        # Script out the drop statements for objects
+                        $options = New-DbaScriptingOption
```

Comment:
Just to keep the pattern going be nice to see this named `dropOptions` since we have a `createOptions` 🧸

Reply:
agreed!